src/circuit/floor_planner/single_pass.rs

```diff
@@ -302,11 +302,15 @@ impl<'r, 'a, F: Field, CS: Assignment<F> + 'a> RegionLayouter<F>
         column: Column<Advice>,
         offset: usize,
         constant: Assigned<F>,
-    ) -> Result<Cell, Error> {
+    ) -> Result<AssignedCell<F, F>, Error> {
         let advice = self.assign_advice(annotation, column, offset, &mut || Ok(constant))?;
         self.constrain_constant(advice, constant)?;
 
-        Ok(advice)
+        Ok(AssignedCell {
+            cell: advice,
+            value: Some(constant.evaluate()),
```

Comment:
This undoes the optimisation that we were trying to make with the `Assigned` type, to enable batching inversions (because this method evaluates the inversion to obtain an `F`). I think the type should be `Option<Assigned<F>>`.